tests/test_pigeon.py

```diff
@@ -67,29 +64,20 @@ def json_data(self):
         ) as fp:
             return fp.read()
 
-    def test_get_and_write_file_data_to_temp(
-        self, mock_osf_api, guid, json_data, file_name
-    ):
-        mock_osf_api.add(
-            responses.GET,
-            f"{settings.OSF_API_URL}v2/guids/{guid}",
-            status=200,
-            body=json_data,
-        )
-
-        with tempfile.TemporaryDirectory() as temp_dir:
-            run(
-                get_and_write_json_to_temp(
+    async def test_dump_json_to_dir(self, guid, json_data, file_name):
+        with aioresponses() as m:
+            m.get(f"{settings.OSF_API_URL}v2/guids/{guid}", body=json_data)
+            with tempfile.TemporaryDirectory() as temp_dir:
+                await dump_json_to_dir(
                     f"{settings.OSF_API_URL}v2/guids/{guid}", temp_dir, file_name
                 )
-            )
-            assert len(os.listdir(temp_dir)) == 1
-            assert os.listdir(temp_dir)[0] == file_name
-            info = json.loads(open(os.path.join(temp_dir, file_name)).read())
-            assert info == json.loads(json_data)
+                assert len(os.listdir(temp_dir)) == 1
```

Comment:
These could/should be outside the `with` context, no?

Reply:
shouldn't matter, but I'm changing it anyway because it looks nicer.